app/src/main/java/ch/epfl/favo/MainActivity.java

```diff
@@ -138,7 +138,9 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
 
     switch (itemId) {
       case R.id.nav_home:
-        navController.navigate(R.id.nav_map);
+        getSupportFragmentManager().popBackStackImmediate();

```

Comment:
What would happen if the backstack has more than 3 fragments? We can probably getFragmentById from the manager and ensure we always get the map view regardless of how deep the backstack is. We can do it on the next sprint

Reply:
I have tried with getFragmentById(), but actually there is no fragment stored in backStack. The strange thing is that popBackStackImmediate() works, it seems it pop the "stack state" rather than "fragment", I haven' t found any method to query "stack state". https://developer.android.com/reference/android/app/FragmentManager#popBackStack()

Reply:
Since confused by these behaviours, I did experiments to check, and found this trick fail to pretend navigation when user continuously click favourPage button more than 3 times, it is relatively rare for a seasonable user...of course there may exist better solution